core/controllers/classifier_test.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the controllers that communicate with VM for training
+classifiers."""
+
+import hashlib
+import hmac
+import json
+import os
+
+from core.domain import classifier_services
+from core.domain import exp_services
+from core.tests import test_utils
+import feconf
+
+
+def get_training_data_from_state(state):
+    """Retrieves training data from the State domain object."""
+    training_data = []
+    for (answer_group_index, answer_group) in enumerate(
+            state.interaction.answer_groups):
+        classifier_rule_spec_index = (
+            answer_group.get_classifier_rule_index())
+        if classifier_rule_spec_index is not None:
+            classifier_rule_spec = answer_group.rule_specs[
+                classifier_rule_spec_index]
+            answers = []
+            for doc in classifier_rule_spec.inputs['training_data']:
+                answers.extend([doc])
```

Comment:
I'm a little confused by the above three lines. How is this different from e.g. `answers = copy.deepcopy(classifier_rule_spec.inputs['training_data'])`?